### ISSUE TITLE
Vagrant looking for remote file - failing - breaking build script and 'vagrant up'

### ISSUE THREAD

**Issue by gametimebrizzle** [rapid7/metasploitable3#309]
## Issue Description
When running 'build_win2008.ps1' I get about 30-45 minutes into the install (shorter after initial download of win2k8 iso) and then I am presented with 

> Build 'virtualbox-iso' errored: Retryable error: Error uploading script: Process exited with status 1

```
==> Some builds didn't complete successfully and had errors:
--> virtualbox-iso: Retryable error: Error uploading script: Process exited with status 1
```

## Host System
* OS: Windows 10
* Packer Version:  1.2.3
* Vagrant Version:  2.1.1
* VirtualBox Version:  5.2.12 r122591 (Qt5.6.2)

## Command Output
```
PS C:\Users\gameTower\metasploitable3> .\build_win2008.ps1
Compatible version of VirtualBox found.
Compatible version of packer found.
Compatible version of Vagrant found.
Compatible version of vagrant-reload plugin found.
All requirements found. Proceeding...
Building the Vagrant box...
virtualbox-iso output will be in this color.

==> virtualbox-iso: Downloading or copying Guest additions
    virtualbox-iso: Downloading or copying: file:///C:/Program%20Files/Oracle/VirtualBox/VBoxGuestAdditions.iso
==> virtualbox-iso: Downloading or copying ISO
    virtualbox-iso: Downloading or copying: http://download.microsoft.com/download/7/5/E/75EC4E54-5B02-42D6-8879-D8D3A25FBEF7/7601.17514.101119-1850_x64fre_server_eval_en-us-GRMSXEVAL_EN_DVD.iso
    virtualbox-iso: Download progress: 0%
    virtualbox-iso: Download progress: 1%
<--snip-->
    virtualbox-iso: Download progress: 100%
==> virtualbox-iso: Creating floppy disk...
    virtualbox-iso: Copying files flatly from floppy_files
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../answer_files/2008_r2/Autounattend.xml
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../scripts/configs/microsoft-updates.bat
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../scripts/configs/win-updates.ps1
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../scripts/installs/openssh.ps1
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../scripts/installs/install_dotnet45.ps1
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../scripts/installs/install_wmf.ps1
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/oracle-cert.cer
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/gdig2.crt
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/comodorsadomainvalidationsecureserverca.crt
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/comodorsacertificationauthority.crt
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/addtrust_external_ca.cer
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/baltimore_ca.cer
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/digicert.cer
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/equifax.cer
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/globalsign.cer
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/gte_cybertrust.cer
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/microsoft_root_2011.cer
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/thawte_primary_root.cer
    virtualbox-iso: Copying file: C:\Users\gameTower\metasploitable3\packer\templates/../../resources/certs/utn-userfirst.cer
    virtualbox-iso: Done copying files from floppy_files
    virtualbox-iso: Collecting paths from floppy_dirs
    virtualbox-iso: Resulting paths from floppy_dirs : []
    virtualbox-iso: Done copying paths from floppy_dirs
==> virtualbox-iso: Creating virtual machine...
==> virtualbox-iso: Creating hard drive...
==> virtualbox-iso: Attaching floppy disk...
==> virtualbox-iso: Creating forwarded port mapping for communicator (SSH, WinRM, etc) (host port xxxx)
==> virtualbox-iso: Executing custom VBoxManage commands...
    virtualbox-iso: Executing: modifyvm metasploitable3-win2k8 --memory 4096
    virtualbox-iso: Executing: modifyvm metasploitable3-win2k8 --cpus 2
==> virtualbox-iso: Starting the virtual machine...
==> virtualbox-iso: Waiting 10m0s for boot...
==> virtualbox-iso: Typing the boot command...
==> virtualbox-iso: Waiting for SSH to become available...
==> virtualbox-iso: Connected to SSH!
==> virtualbox-iso: Uploading VirtualBox version info (5.2.12)
==> virtualbox-iso: Uploading VirtualBox guest additions ISO...
==> virtualbox-iso: Uploading C:\Users\gameTower\metasploitable3\packer\templates/../../scripts => C:/vagrant
==> virtualbox-iso: Uploading C:\Users\gameTower\metasploitable3\packer\templates/../../resources => C:/vagrant
==> virtualbox-iso: Provisioning with shell script: C:\Users\gameTower\metasploitable3\packer\templates/../../scripts/configs/update_root_certs.bat
==> virtualbox-iso: Deregistering and deleting VM...
==> virtualbox-iso: Deleting output directory...
Build 'virtualbox-iso' errored: Retryable error: Error uploading script: Process exited with status 1

==> Some builds didn't complete successfully and had errors:
--> virtualbox-iso: Retryable error: Error uploading script: Process exited with status 1

==> Builds finished but no artifacts were created.
Error building the Vagrant box using Packer. Please check the output above for any error messages.

At C:\Users\gameTower\metasploitable3\build_win2008.ps1:108 char:9
+         throw "Error building the Vagrant box using Packer. Please ch ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Error building ...error messages.:String) [], RuntimeException
    + FullyQualifiedErrorId : Error building the Vagrant box using Packer. Please check the output above for any error messages.
```

Here, I try running the next command in 'build_win2008.ps1' to see what happens:	
	
```
PS C:\Users\gameTower\metasploitable3> cmd.exe /c vagrant box add metasploitable3-win2k8 packer\windows_2008_r2_virtualbox.box
	The box 'packer\windows_2008_r2_virtualbox.box' could not be found or
	could not be accessed in the remote catalog. If this is a private
	box on HashiCorp's Vagrant Cloud, please verify you're logged in via
	`vagrant login`. Also, please double-check the name. The expanded
	URL and error message are shown below:

	URL: ["https://vagrantcloud.com/packer/windows_2008_r2_virtualbox.box"]
	Error: The requested URL returned error: 404 Not Found
```

-The script is looking for a prebuilt box file from VagrantCloud?  Why?
-Note: https://vagrantcloud.com/packer/windows_2008_r2_virtualbox.box is, in fact, NOT a resolvable URL.

Looking into this error:

>  "Error building the Vagrant box using Packer. Please check the output above for any error messages." 

The build script[_**build_win2008.ps1**_] is apparently failing, but it tells me just about nothing at all, except the build failed.  

Here is the snippet from **buld_win2008.ps1**
```

if($?) {
        Write-Host "Box successfully built by Packer."
    } else {
        throw "Error building the Vagrant box using Packer. Please check the output above for any error messages."
    }
```
	
The 'throw' here results in the deletion of about 30 minutes of build time, I'm assuming 'update_root_certs.bat' is failing? 

```
==> virtualbox-iso: Provisioning with shell script: C:\Users\gameTower\metasploitable3\packer\templates/../../scripts/configs/update_root_certs.bat
==> virtualbox-iso: Deregistering and deleting VM...
==> virtualbox-iso: Deleting output directory...
```

**However the metasploitable3-win2k8-disk001.vmdk (below) can be used to bring up Metasploitable-Win2k8 in VirtualBox**

```
PS C:\Users\gameTower\.vagrant.d\boxes\metasploitable3\0\virtualbox> ls


    Directory: C:\Users\gameTower\.vagrant.d\boxes\metasploitable3\0\virtualbox


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        6/17/2018   3:13 PM          10651 box.ovf
-a----        6/17/2018   3:13 PM             26 metadata.json
-a----        6/22/2018   2:22 PM     6871455232 metasploitable3-win2k8-disk001.vmdk
-a----        6/17/2018   3:13 PM           1961 Vagrantfile
```

I try again after adding _C:\users\gameTower\.vagrant.d_ to **VAGRANT_HOME**...just to see what happens

```
PS C:\users\gameTower\metasploitable3> vagrant up
Bringing machine 'ub1404' up with 'virtualbox' provider...
Bringing machine 'win2k8' up with 'virtualbox' provider...
==> ub1404: Box 'metasploitable3-ub1404' could not be found. Attempting to find and install...
    ub1404: Box Provider: virtualbox
    ub1404: Box Version: >= 0
==> ub1404: Box file was not detected as metadata. Adding it directly...
==> ub1404: Adding box 'metasploitable3-ub1404' (v0) for provider: virtualbox
    ub1404: Downloading: metasploitable3-ub1404
    ub1404:
An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

Couldn't open file /C:/users/gameTower/metasploitable3/metasploitable3-ub1404
```
Try with adding win2k8 specification, but still no luck.
```
PS C:\users\gameTower\metasploitable3> vagrant up win2k8
Bringing machine 'win2k8' up with 'virtualbox' provider...
==> win2k8: Box 'metasploitable3-win2k8' could not be found. Attempting to find and install...
    win2k8: Box Provider: virtualbox
    win2k8: Box Version: >= 0
==> win2k8: Box file was not detected as metadata. Adding it directly...
==> win2k8: Adding box 'metasploitable3-win2k8' (v0) for provider: virtualbox
    win2k8: Downloading: metasploitable3-win2k8
    win2k8:
An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

Couldn't open file /C:/users/gameTower/metasploitable3/metasploitable3-win2k8

```

I'm not very familiar with provisioning in general, so can someone help me out here?  I have a working metasploitable3-win2k8.vbox, but the machine isn't provisioned to be vulnerable as intended.

**Comment by gametimebrizzle** [comment]
nothing?

What else do I need to provide?

Which windows logs do I need to include?

Should I post my vboxHardening.log?

**Comment by gametimebrizzle** [comment]
I guess my questions are:
- Why is Vagrant not provisioning my packer-built Win2k8 file (which is .vbox, not .box)
-  Can I provison this manually? I.e, batch the batch scripts together? I assume I'd run all provisioning bat file on the win2k8 vbox?

**Comment by TheLinuxNinja** [comment]
Looks like it failed in the middle of uploading the .bat files.
I was able to get this to build successfully on my Mac, about to try it on my Arch Linux workstation. I haven't tried it on Windows as I don't have a Windows machines to test on.

It seems like you have some vagrant machines left behind.

Run:
`vagrant destroy`

and make sure the machine isn't still in VirtualBox.

Then start from the beginning and see if it fails in the same step.

**Comment by jmartin-r7** [comment]
Please pull and test again #277 may help with this.

**Comment by jmartin-r7** [comment]
Closing as expected to have been fixed with #277.
